src/Sentry/Internal/Hub.cs

```diff
@@ -188,9 +214,18 @@ public SentryId CaptureEvent(SentryEvent evt, Scope? scope = null)
                     evt.Contexts.Trace.ParentSpanId = linkedSpan.ParentSpanId;
                 }
 
+                // Treat all events as errors
+                _sessionManager.ReportError();
+
                 var id = currentScope.Value.CaptureEvent(evt, actualScope);
                 actualScope.LastEventId = id;
 
+                // If the event contains unhandled exception - end session as crashed
+                if (evt.SentryExceptions?.Any(e => !(e.Mechanism?.Handled ?? true)) ?? false)
+                {
+                    EndSession(SessionEndStatus.Crashed);
+                }
```

Comment:
Is this good enough?

Reply:
I think this will do!
Lets worry about that native crash story later on. As we chat on the call we might turn off the .NET session support and use the native for Android and iOS anyway for MAUI, Unity at least